src/uct/tcp/tcp_iface.c

```diff
@@ -481,35 +481,43 @@ static void uct_tcp_iface_ep_list_cleanup(uct_tcp_iface_t *iface,
     uct_tcp_ep_t *ep, *tmp;
 
     ucs_list_for_each_safe(ep, tmp, ep_list, list) {
+        uct_tcp_cm_purge_ep(ep);
```

Comment:
is there assert in ep destroy internal that ep caps are 0?

Reply:
yep, it was added in af10e8f4ce80ff30d582d0a85959aa2f83511590 commit:
```
static UCS_CLASS_CLEANUP_FUNC(uct_tcp_ep_t)
{
    uct_tcp_iface_t UCS_V_UNUSED *iface =
        ucs_derived_of(self->super.super.iface, uct_tcp_iface_t);

    if (self->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_TX)) {
        uct_tcp_ep_remove_ctx_cap(self, UCT_TCP_EP_CTX_TYPE_TX);
    }

    if (self->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_RX)) {
        uct_tcp_ep_remove_ctx_cap(self, UCT_TCP_EP_CTX_TYPE_RX);
    }

    ucs_assertv(!self->ctx_caps, "ep=%p", self);                   <=====

    ...
}
```